test/e2e-go/cli/goal/expect/goalNodeTest.exp

```diff
@@ -36,6 +36,24 @@ if { [catch {
     # Stop node
     ::AlgorandGoal::StopNode $TEST_PRIMARY_NODE_DIR
 
+    # "break" the node by replacing it's ledger data files with "broken" ones.
+    lassign [exec find $TEST_PRIMARY_NODE_DIR -name "ledger.tracker.sqlite"] PRIMARY_TRACKER_DATABASE_FILE
+    exec find $TEST_PRIMARY_NODE_DIR -name "ledger.tracker.sqlite*" -delete
+    exec -- echo "1234" > $PRIMARY_TRACKER_DATABASE_FILE
+
+    # try to start the primary node, and observe the expected failuire
+    set timeout 15
+    spawn goal node start -d $TEST_PRIMARY_NODE_DIR
+    expect {
+        timeout { close; ::AlgorandGoal::Abort "starting node exceeded timeout" }
+        "^Algorand node failed to start: node exited with an error code, check node\.log for more details : exit status 1" {puts "Expected failuire : node failed to start"; close}
```

Comment:
should add the eof check line: 
`	eof { ::AlgorandGoal::Abort "eof received before the expected output " }`
Otherwise, the eof will match when the `"^Algorand node failed to start...` is not matched, and proceed without reporting an error.

Reply:
added.